contracts/base/FunctionHandlerManager.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import {ISafeProtocolFunctionHandler} from "../interfaces/Integrations.sol";
+import {ISafe} from "../interfaces/Accounts.sol";
+import {RegistryManager} from "./RegistryManager.sol";
+
+/**
+ * @title FunctionHandlerManager
+ * @notice This contract manages the function handlers for the Safe Account. The contract stores the
+ *        information about Safe account, bytes4 function selector and the function handler contract address.
+ * @dev This contract inherits BaseManager so that `onlyPermittedIntegration` modifier can be used.
+ */
+abstract contract FunctionHandlerManager is RegistryManager {
+    // Storage
+    /** @dev Mapping that stores information about Safe account, function selector, and address of the account.
+     */
+    mapping(address => mapping(bytes4 => address)) public functionHandlers;
+
+    // Events
+    event FunctionHandlerChanged(address indexed safe, bytes4 indexed selector, address indexed functionHandler);
```

Comment:
do we need all of the three indexed?

Reply:
I think all 3 would be helpful to query:
- Distinct function handlers
- function handler(s) used for specific selectorId
- Safes with function handlers enabled

So, kept marked these as indexed